### ISSUE TITLE
The phenomena of loss convergence but ATE oscillation

### ISSUE THREAD

**Issue by nyzhyxydsh** [Sachini/ronin#38]
Thank you for your dataset and code. After several times, we found some common problems and came here for advice.
After the convergence of the loss of train, then test work were carried out on each epoch(we run every 
checkpoint on the  "seen_subjects_test_set"), and we find that ATE did not converge and a continuous oscillation curve appeared. This phenomenon appears in both R-RESNet and R-LSTM.
I want to know if it's normal to this phenomenon  (no code changes in github)? I'm always wondering  in this case the ATE doesn't converge and how can i get the best ATE?
thanks.
Some of the data information during the experiment are followings:
The left figure show the ATE  oscillation phenomenon around checkpoint 300, where the difference exceeds 0.5.
The middle figure show the training loss diagram of LSTM model ,train epoch =800.
The right figure  show the ATE test results of the test set at each checkpoint.

![G_67W LZV5LAT~73XTT5%_Q](https://user-images.githubusercontent.com/49915101/148419995-812ffc5d-dbf1-42bd-a749-6ba6d94f2688.png)

**Comment by Sachini** [comment]
This is consistent with our observations, assuming ATE calculation here is for each full trajectory. I believe this is because we rely on the game rotation vector and does not explicitly handle drift correction.  You can check [TLIO](https://arxiv.org/abs/2007.01867) or (IDOL)[https://arxiv.org/abs/2102.04024] which extends this framework to explicitly calculate drift-free orientations. 

(Feel free to open the issue if further discussion)
